wordpress-mobile/WordPress-iOS
# Reader: Refactor filter sheet

Closes #22211

> [!NOTE]
> This merges into `feature/reader-navigation-filter`.

This PR covers some groundwork changes for the upcoming filter pill buttons and updates the look of the Filter sheet. My original work in progress has become too large and I decided to split the changes, as I'm also in the middle of rebasing my ongoing work on top of your changes at #22260.

Included in this PR: 

- Refactor `FilterSheetViewController`; merge `FilterSheetView` into the view controller and remove any tab management code.
- Update the cell style for 'filter by blog' and 'filter by tag'. ⚠️ **Note that this excludes the Edit button.** I'll add it through a separate PR.
- Move the `FilterProvider` management to the tab view model.
- Show a simple Text element to indicate how many filters are available for the current stream.
- Rename `shouldHideButtons` to `shouldHideStreamFilters` to fit better with the new navigation hierarchy.
---

Apart from this change, in summary, here's how the filter implementation is going to change (which would be covered in another 2-3 follow-up PRs):

- The filter sheet will only hold one instance of `FilterProvider`. 
- The `FilterProvider`s will serve as the model for the filter buttons—Imagine pulling out the tabs within the current Filter sheet and turning them into the new filter buttons. 
- Previously, the filter sheet fetches the list of blogs (or tags) _after_ the sheet is loaded. In the new design, this needs to happen every time the user switches the stream. 

## To test

- Launch the Jetpack app and go to Reader.
- The filter sheet is now temporarily placed in the Settings button. Tap the Settings button while in the Following tab to open the filter sheet for tags.
- 🔎  Verify that the filter sheet appears and shows 'Filter by tag'.
- 🔎  Verify that selecting the tag filters the current stream.
- Switch to an organization tab and tap the Settings button.
- 🔎  Verify that the filter sheet appears and shows 'Filter by blog'.
- 🔎  Verify that selecting the blog filters the current stream.

> [!WARNING]
> There's currently no way to reset the filter for now as the entry point is just a makeshift to test the sheet implementation. I'll remove the test code when this is approved.

## Regression Notes
1. Potential unintended areas of impact
Should be none. The feature is unreleased and there will be further tests on the main feature branch `feature/reader-navigation-filter`.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manually tested the changes.

3. What automated tests I added (or what prevented me from doing so)
N/A.

PR submission checklist:


UI Changes testing checklist:
